Add Posit Assistant vs. Claude Code blog post (#288)
posit-dev/open-source-website: content/blog/comparing-posit-assistant-and-claude-code/index.md

---
title: "Comparing Posit Assistant and Claude Code"

> date: 2026-06-02

Author: ```suggestion
date: 2026-06-08
```

image: "images/featured.png"
image-alt: "Posit Assistant logo with 'Posit Assistant' and 'Claude Code' labels on a light purple background."
topics:
  - Artificial Intelligence
software: []
languages:
  - R
  - Python
hidesubscription: false
tags:
  - posit-assistant
---

We often get questions about how Posit Assistant compares to Claude Code. If you're already using Posit Assistant with a Claude model, are they really that different? At a high-level, Claude Code is a general-purpose coding agent, but **we designed Posit Assistant specifically for people who work with data.** This post covers three specific differences: 

1. Posit Assistant has out-of-the-box access to your R or Python session. 
2. Posit Assistant can display plots in the chat, leading to easier interpretation for you and the agent. 
3. Posit Assistant has specific interaction modes, skills, and prompting for data analysis. 

You can also see a full demo of the differences between the two in this video:

{{< video src="https://www.youtube.com/watch?v=7GI6-4J0AXA" width="600" >}} 

<div class="callout callout-tip" role="note" aria-label="Tip">
<div class="callout-header">

> <span class="callout-title">What's the difference between Posit Assistant, Positron Assistant, and Posit AI?</span>

Reviewer: NICE

</div>
<div class="callout-body">

> This post talks about Posit Assistant, Posit's latest coding agent available in RStudio and Positron. Posit AI is a model subscription service that can power Posit Assistant. Positron Assistant is a coding agent built into Positron. We plan for Posit Assistant to supersede Positron Assistant in Q3 2026. 

Author: ```suggestion
This post talks about [Posit Assistant](https://posit-dev.github.io/assistant/), Posit's latest coding agent available in RStudio and Positron. [Posit AI](https://posit.co/products/ai) is a model subscription service that can power Posit Assistant. [Posit*ron* Assistant](https://positron.posit.co/assistant.html) is a coding agent built into Positron. We plan for Posit Assistant to supersede Positron Assistant in Q3 2026. 
```

</div>
</div>

### Coding agents vs. chat apps

> Claude Code and Posit Assistant are both _coding agents_. An _agent_ is a ["model running tools in a loop"](https://simonwillison.net/2025/Sep/18/agents/), meaning an agent can see your environment, take actions in that environment (e.g., run code), and iterate based on the outcome of those actions. These abilities make agents vastly more useful for coding than chat apps like ChatGPT. 

Author: ```suggestion
Claude Code and Posit Assistant are both _coding agents_. An _agent_ is a ["model running tools in a loop,"](https://simonwillison.net/2025/Sep/18/agents/) meaning an agent can see your environment, take actions in that environment (e.g., run code), and iterate based on the outcome of those actions. These abilities make agents vastly more useful for coding than chat apps like ChatGPT. 
```

2. You might not want to create and run scripts for every bit of code. 
3. It leaves you and the agent on different footing: you have direct access to your session, but Claude Code doesn't. This can make errors more likely and the general experience more frustrating. 

It is possible to give Claude Code access to your R or Python session by using an MCP server. In the walkthrough video, we show how to do this using the [btw package](https://github.com/posit-dev/btw), providing Claude Code the ability to access our R session and run R code directly. 

Even with an MCP server, however, Claude Code's interface can make it difficult to see the code being run on your behalf. For some tasks, this might not matter, but if you're analyzing data it can be important to understand the analysis in detail. For data analysis tasks, Posit Assistant is designed for auditability and transparency. The code is visible in the tool call UI with syntax highlighting and styling, so you can easily audit what's happening to your data at each step.

> See this comparison in the video: [Claude Code](https://youtu.be/7GI6-4J0AXA?t=220) | [Posit Assistant](https://youtu.be/7GI6-4J0AXA?t=424)

Reviewer: Really like these timestamped links

Posit Assistant is both a general-purpose coding agent and a specialized data analysis agent, and it behaves slightly differently depending on your task. During data exploration, it only runs a few bits of code at a time, then summarizes what it found and suggests next steps. Because the purpose of data analysis, especially exploration, is for you to learn about the data, the entire process breaks down if the model runs ahead with analysis you can't keep up with. **Posit Assistant's approach to co-analysis with the user is designed to support your understanding of the data rather than hindering it.** 

<figure>
<img src="images/pa-data-analysis.gif" alt="GIF of Posit Assistant carrying out some data analysis on the likeliness words data. Posit Assistant does three rounds of tool calls, summarizes what it found, and provides some suggestions." width="600px" />
<figcaption>Posit Assistant explores data iteratively, running a few bits of code at a time before summarizing and suggesting next steps.</figcaption>
</figure>

> Claude Code excels at writing code, but is oriented towards writing code to files, running, and reviewing, rather than exploring data step-by-step with the user. 

Author: ```suggestion
Claude Code excels at writing code, but is oriented towards writing code to files, executing it, and reviewing, rather than exploring data step-by-step with the user. 
```

**Posit Assistant also has specialized prompting, skills, and tools that promote sound data analysis.** These include skills for making Shiny apps and Quarto reports, a [data cleaning mode](https://opensource.posit.co/blog/2026-05-08_ai-newsletter/), and prompting about rigorous statistical and modeling practices.

![Screenshot of Posit Assistant's data cleaning mode, showing a tabbed interface with issues like 'Sparse early/late years'. The assistant presents a table of row counts by year range, explains the imbalance, and offers actionable choices: keep all years, filter to 2011–2022, or filter to 2011–2019.](images/data-cleaning-mode.png "Posit Assistant in data cleaning mode, surfacing questions to the user.") 

See this comparison in the video: [Claude Code](https://youtu.be/7GI6-4J0AXA?t=990) | [Posit Assistant](https://youtu.be/7GI6-4J0AXA?t=1133)

## Posit Assistant is designed for data work

Claude Code, Codex, and other coding agents are very useful tools. They excel at general-purpose coding tasks, and you might want to use them for work that benefits from running multiple agents in parallel or to take advantage of your existing subscription plans.  

> However, we made Posit Assistant because we think people who work with data should have an excellent agent specifically designed for them and their needs. This includes data analysis and visualization, but also package development, Shiny app creation, and other coding-focused tasks in the data science ecosystem.

Author: ```suggestion
However, we made Posit Assistant because we think people who work with data should have an excellent agent specifically designed for them and their needs. Those needs include data analysis and visualization, but also package development, Shiny app creation, and other coding-focused tasks in the data science ecosystem.
```
